src/copilot/app/rag/rag_processor.py

```diff
@@ -42,38 +52,41 @@ async def retrieve(self, request: RAGRequest, language: str = None, k: int = 0):
 
         Parameters
         ----------
+        db : Session
+            Database session
         request : RAGRequest
             User input question
         language : str
             Question and context documents language
         k : int, default 0
             Number of context documents to return (need to be revised, current logic in the code is confusing)
         """
-        k = self.k_retrieve if k is None else k
+        rows = document_service.get_semantic_match(db, request.query, language=language, k=k)
 
-        rows = await semantic_similarity_match(request.query, db_name='embeddings', language=language, k=k)
-        documents = [dict(row) for row in rows][0]
+        return rows[0] if len(rows) > 0 else {"text": "", "url": ""}
 
-        return {"contextDocs": documents["text"], "sourceUrl": documents["url"], "cosineSimilarity": documents["similarity_metric"]}
-
-    async def process(self, request: RAGRequest):
+    def process(self, db: Session, request: RAGRequest, language: str = None):
         """
         Execute the RAG process and query the LLM model.
 
         Parameters
         ----------
+        db : Session
+            Database session
         request : RAGRequest
             User input question
+        language : str
+            Question and context documents language
 
         Returns
         -------
         str
             LLM generated answer to the question
         """
-        documents = await self.retrieve(request)
-        context_docs = documents['contextDocs']
-        source_url = documents['sourceUrl']
-        messages = self.create_openai_message(context_docs, request.query)
+        documents = self.retrieve(db, request, language=language, k=self.k_retrieve)
+        context_doc = documents.text
```

Comment:
Follow-up: using the indexing/embed_rag_data route and changing to `documents['text']` instead of `documents.text` (same for `url` key) solves the bug, but not documents are retrieved (or empty).